core/renderers/zelos/info.js

```diff
@@ -376,16 +377,16 @@ Blockly.zelos.RenderInfo.prototype.finalizeOutputConnection_ = function() {
       this.outputConnection.shape.connectionOffsetX(connectionWidth);
 
   // Adjust right side measurable.
-  if (!this.hasStatementInput) {
+  this.startX = connectionWidth;
+  if (!this.hasStatementInput && !this.bottomRow.hasNextConnection) {
     this.rightSide.height = connectionHeight;
     this.rightSide.width = connectionWidth;
     this.rightSide.centerline = connectionHeight / 2;
     this.rightSide.xPos = this.width + connectionWidth;
+    connectionWidth *= 2;
```

Comment:
Could we add some comments for this section? For example why are we multiplying connectionWidth by 2? And why do we care if it has a next connection or not?

Reply:
Okay and will simplify the logic to make it clearer.